server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java

```diff
@@ -1186,7 +1192,11 @@ protected void transferToResult(IMNode node) {
         new MNodeCollector<List<PartialPath>>(root, pathPattern) {
           @Override
           protected void transferToResult(IMNode node) {
-            resultSet.add(node.getPartialPath());
+            try {
+              resultSet.add(getCurrentPartialPath(node));
+            } catch (MetadataException e) {
+              logger.error(e.getMessage());
```

Comment:
Should we throw exception again here?

Reply:
I'm not sure how to deal with this. It is a exception will not happen actually, so I did not throw it again here.
The exception is caused by a construct of ParitialPath, the content is nodes during a traverse which should assure to avoid illegal path. But this method override an abstract method which does not throw exception, so this catch did not throw again.